src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/generalapplicationdirections/GeneralApplicationDirectionsNewEventAboutToSubmitHandlerTest.java

```diff
@@ -456,6 +460,32 @@ void givenCase_whenApproveAnApplication_thenUpdateStatusApprovedCompletedAndRetu
         verify(gaDirectionService).submitCollectionGeneralApplicationDirections(any(), any(), any());
     }
 
+    /*
+     * Opted for reflection to test performAddHearingIfNecessary.  Avoids complex setup to test this private method.
+     * Test written to ensure that ManageHearingsAction.ADD_HEARING is set by performAddHearingIfNecessary.
+     */
+    @Test
+    void givenCase_whenPerformAddHearingIfNecessary_thenCaseWithHearingHasAddHearingAction()
+        throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+
+        // Arrange
+        FinremCaseDetails caseDetails = buildFinremCallbackRequest().getCaseDetails();
+        when(gaDirectionService.isHearingRequired(caseDetails)).thenReturn(true);
+        String userAuthorisation = "Bearer test-token";
+
+        // Act, call performAddHearingIfNecessary using reflection
+        Method method = GeneralApplicationDirectionsNewEventAboutToSubmitHandler.class.getDeclaredMethod(
```

Comment:
Do you really need to do this? From the simplicity of what the method does why does it need complex setup that isn't required in all the other tests?